### PR TITLE
New ups type

### PR DESCRIPTION
**Pull request by kevin-hereworks**

### CODE REVIEW

**Comment by kevin-hereworks** [comment]
Hi, not sure on the best way to add units for battery_energy_capacity, units are ampere_hours but not sure whether this should be under energy density or under a new batterycapacity measurement

File: ontology/yaml/resources/units/units.yaml [hunk 1]
```diff
@@ -49,6 +49,11 @@ area:
   square_inches:
     multiplier: 0.00064516
     offset: 0
+batterycapacity:
```
Comment (cstirdivant) [inline ontology/yaml/resources/units/units.yaml:52]: This is not a valid measurement type. See note above.

File: ontology/yaml/resources/subfields/subfields.yaml [hunk 2]
```diff
@@ -261,6 +261,7 @@ measurement:
   angularmomentum: "The quantity of rotation of a body, which is the product of its moment of inertia and its angular velocity."
   aqi: "Air quality index, a set of special standard gradings for the pollution in air."
   area: "Amount of space within the perimeter of a 2D shape"
+  batterycapacity: "Energy charge in a battery"
```
Comment (cstirdivant) [inline ontology/yaml/resources/subfields/subfields.yaml:264]: This is not a valid measurement type. Please use `electriccharge`

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 3]
```diff
@@ -3102,3 +3102,7 @@ literals:
 - ultraviolet_irradiance_sensor:
     fixed_min: 0.0
     flexible_max: 2000.0
+
+- batterycapacity_sensor:
```
Comment (cstirdivant) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:3106]: Please fix based on subfield notes.

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml [hunk 4]
```diff
@@ -133,6 +152,16 @@ BTEPM:
   - battery_current_sensor
   - line_frequency_sensor  
 
+SBATM:
+  description: "Simple battery monitoringy typically used with UPS"
+  is_abstract: true
+  uses:
+  - battery_charge_percentage_sensor
+  - batterycapacity_sensor
```
Comment (cstirdivant) [inline ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml:160]: Please fix based on subfields notes.

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml [hunk 5]
```diff
@@ -133,6 +152,16 @@ BTEPM:
...
+  - battery_status
+  - battery_test_status
+  - failed_battery_alarm
```
Comment (cstirdivant) [inline ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml:163]: `alarm` fields should always be `opt_uses`

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml [hunk 6]
```diff
@@ -133,6 +152,16 @@ BTEPM:
   - battery_current_sensor
   - line_frequency_sensor  
 
+SBATM:
+  description: "Simple battery monitoringy typically used with UPS"
```
Comment (cstirdivant) [inline ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml:156]: This is not a valid abstract type, I would suggest adding fields as `opt_uses` to the existing `BATM` type.

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml [hunk 7]
```diff
@@ -63,6 +63,21 @@ IOBM:
...
+  - output_phase1_neutral_voltage_sensor
+  - output_phase2_neutral_voltage_sensor
+  - output_phase3_neutral_voltage_sensor
+  - output_phase1_current_sensor
+  - output_phase2_current_sensor
+  - output_phase3_current_sensor
+  - output_frequency_sensor
+  - output_power_sensor
+  - output_apparent_power_sensor
+  - bypass_status
```
Comment (cstirdivant) [inline ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml:79]: Suggest making this field optional, see optional fields on `IOBM`

**Review by cstirdivant — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/subfields/subfields.yaml (hunk 2 above)
Comment (kevin-hereworks) [reply ontology/yaml/resources/subfields/subfields.yaml]: fixed thank you

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml (hunk 5 above)
Comment (kevin-hereworks) [reply ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml]: fixed thank you

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml (hunk 6 above)
Comment (kevin-hereworks) [reply ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml]: removed

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml (hunk 7 above)
Comment (shambergoldstein) [reply ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml:81]: Please move frequency, power, and apparent power fields to optional as well to match `IOBM`.

File: ontology/yaml/resources/units/units.yaml [hunk 8]
```diff
@@ -101,6 +101,11 @@ distance:
     offset: 0
 electriccapacitance:
   farads: STANDARD
+electriccharge:
```
Comment (cstirdivant) [inline ontology/yaml/resources/units/units.yaml:104]: The standard unit is actually coulomb

File: ontology/yaml/resources/units/units.yaml [hunk 9]
```diff
@@ -101,6 +101,11 @@ distance:
...
+  milliampere_hours: STANDARD
+  ampere_hours:
+    multiplier: 1000
```
Comment (cstirdivant) [inline ontology/yaml/resources/units/units.yaml:107]: Please revise this multiplier to convert from the standard unit of coulomb to whatever target unit you need.

File: ontology/yaml/resources/subfields/subfields.yaml [hunk 10]
```diff
@@ -262,6 +262,7 @@ measurement:
   angularmomentum: "The quantity of rotation of a body, which is the product of its moment of inertia and its angular velocity."
   aqi: "Air quality index, a set of special standard gradings for the pollution in air."
   area: "Amount of space within the perimeter of a 2D shape"
+  electriccharge: "Energy charge in a battery"
```
Comment (cstirdivant) [inline ontology/yaml/resources/subfields/subfields.yaml:265]: Please revise this to be more general. For example: "The amount of energy or electrons that pass from one body to another by different modes like conduction, induction or other specific methods (e.g., the charge stored in a battery)."

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml [hunk 11]
```diff
@@ -106,6 +124,10 @@ PPWM:
   guid: "aab62bba-a92f-47bf-bf14-dcb8e61ac7a8"
   description: "Phase power monitoring"
   is_abstract: true
+  opt_uses:
```
Comment (cstirdivant) [inline ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml:127]: Please create a new type for these power fields (see `METERS/PLPM` for a good example)

**Review by cstirdivant — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml (hunk 11 above)
Comment (kevin-hereworks) [reply ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml:127]: Do you want me to create a new type for just `- phase1_power_sensor`,  `- phase2_power_sensor`,  `- phase3_power_sensor`?

Or remove these from PPWM and make a copy of PLPM here?

The PLPM has the apparant power and the power fields in the meters

Comment (cstirdivant) [reply ontology/yaml/resources/ELECTRICAL/entity_types/ABSTRACT.yaml:127]: Please create a new type in this file called `PLPM` and it only needs to have the three power_sensor fields as required (you can put the apparent_power fields and the `line_frequency_sensor` in the opt_uses section on the `PLPM` type).

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 12]
```diff
@@ -3129,6 +3129,10 @@ literals:
     fixed_min: 0.0
     flexible_max: 2000.0
 
+- electriccharge_sensor:
+    fixed_min: 0.0
+    flexible_max: 10000.0
```
Comment (cstirdivant) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:3134]: These values need to be in the STANDARD units as defined in the units.yaml. I am assuming you need to convert this to coulombs.

**Review by cstirdivant — CHANGES_REQUESTED** [review]

**Pull request closed without merging**